sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-processing-steps/apply-stop-bc-funding-restriction-step.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from "@nestjs/common";
+import { EntityManager } from "typeorm";
+import {
+  Restriction,
+  RestrictionActionType,
+  OfferingIntensity,
+} from "@sims/sims-db";
+import { ECertProcessStep } from "./e-cert-steps-models";
+import { ProcessSummary } from "@sims/utilities/logger";
+import { EligibleECertDisbursement } from "../disbursement-schedule.models";
+import { BC_FUNDING_TYPES } from "@sims/services/constants";
+
+/**
+ * Handles BC funding restrictions for both full-time and part-time students.
+ * Applies restrictions that stop BC funding from being disbursed.
+ */
+@Injectable()
+export class ApplyStopBCFundingRestrictionStep implements ECertProcessStep {
+  // Mapping of offering intensity to corresponding restriction action type
+  private readonly restrictionMap = {
+    [OfferingIntensity.fullTime]: RestrictionActionType.StopFullTimeBCFunding,
+    [OfferingIntensity.partTime]: RestrictionActionType.StopPartTimeBCFunding,
+  };
+
+  executeStep(
+    eCertDisbursement: EligibleECertDisbursement,
+    _entityManager: EntityManager,
+    log: ProcessSummary,
+  ): boolean {
+    const offeringIntensity = eCertDisbursement.offering.offeringIntensity;
+    const restrictionType = this.restrictionMap[offeringIntensity];
+
+    // Skip if no matching restriction type for the offering intensity
+    if (!restrictionType) {
+      return true;
+    }
```

Comment:
An unknown or undefined offering intensity would be a huge issue that is unlikely to happen.
The check could be removed IMO. If there is a strong desire to keep it, it should throw an error.